p2p/conn/connection.go

```diff
@@ -200,15 +202,56 @@ func (c *MConnection) OnStart() error {
 	return nil
 }
 
-// OnStop implements BaseService
-func (c *MConnection) OnStop() {
+// FlushStop replicates the logic of OnStop,
+// but flushes all pending msgs before closing the connection.
```

Comment:
// This method ensures that all successful .Send() calls will get flushed before closing